providers/account.rb

```diff
@@ -234,3 +239,23 @@ def group_resource(exec_action)
       new_resource.updated_by_last_action(true) if r.updated_by_last_action?
   end
 end
+
+def keypair_resource(exec_action)
+  new_resource.ssh_keypair.each do |name, key|
+    # avoid variable scoping issues in resource block
+    key_name, key_content = name, key
+
+    home = Etc.getpwnam(new_resource.username).dir
```

Comment:
@chr4 sorry for the delay in getting back to you. Been a busy few weeks. 😒 

Looking around this file I think you might be able to work around this by using the `@my_home` instance variable. It seems like this is used in a few other places where directory creation is handled. I wonder if it snuck in while this PR was sitting open.

Here are some examples of it being used:

* https://github.com/chr4-cookbooks/user/blob/e4959685d6e5db96939c0fb0e6f51bc21901437d/providers/account.rb#L145
* https://github.com/chr4-cookbooks/user/blob/e4959685d6e5db96939c0fb0e6f51bc21901437d/providers/account.rb#L201